Python/Storage/README.md

```diff
@@ -1,21 +1,70 @@
+http://blogs.technet.com/b/windowshpc/archive/2015/04/16/linux-blob-transfer-python-code-sample.aspx
 
-The blobxfer script is a python script that can be used on any platform where a modern Python interpreter can be installed. The script requires two prerequisite packages to be installed: (1) azure and (2) requests. The azure package is required for the script to utilize the Azure Python SDK to interact with Azure using a management certificate or a shared key. The requests package is required for SAS support. If SAS is not needed, one can remove all of the requests references from the script to reduce the prerequisite footprint. You can install these packages using pip, easy_install or through standard setup.py procedures.
+The blobxfer.py script allows interacting with storage accounts using any of
+the following methods: (1) management certificate, (2) shared account key,
+(3) SAS key. The script can, in addition to working with single files, mirror
+entire directories into and out of containers from Azure Storage, respectively.
+Block- and file-level MD5 checksumming for data integrity is supported along
+with various transfer optimizations, built-in retries, and user-specified
+timeouts.
 
-Program parameters and command-line options can be listed via the -h switch. At the minimum, three positional arguments are required: storage account name, container name, local resource. Additionally, one of the following authentication switches must be supplied: --subscriptionid with --managementcert, --storageaccountkey, or --saskey. It is recommended to use SAS keys wherever possible, only HTTPS transport is used in the script.
+The blobxfer script is a python script that can be used on any platform where
+a modern Python interpreter can be installed. The script requires two
+prerequisite packages to be installed: (1) azure and (2) requests. The azure
+package is required for the script to utilize the Azure Python SDK to interact
+with Azure using a management certificate or a shared key. The requests package
+is required for SAS support. If SAS is not needed, one can remove all of the
+requests references from the script to reduce the prerequisite footprint. You
+can install these packages using pip, easy_install or through standard
+setup.py procedures.
 
-The script will attempt to perform a smart transfer, by detecting if the local resource exists. For example:
+Program parameters and command-line options can be listed via the -h switch. At
+the minimum, three positional arguments are required: storage account name,
+container name, local resource. Additionally, one of the following
+authentication switches must be supplied: --subscriptionid with
+--managementcert, --storageaccountkey, or --saskey. It is recommended to use
+SAS keys wherever possible; only HTTPS transport is used in the script.
 
+The script will attempt to perform a smart transfer, by detecting if the local
+resource exists. For example:
```

Comment:
Good point. I will clarify. The script will not proactively look on the remote end since this incurs network cost, if it detects a local file, then it will assume upload.